### PR TITLE
[FLINK-11251] [metrics] GenericValueMetricGroup should always ignore its group name in logical scope

### PR DESCRIPTION
**Pull request by tony810430**
## What is the purpose of the change

This PR fixes an issue in the metric system where value from user-defined variable wasn't ignored in logic scope in some situations. As a result, some reporters using logic scope as metric name, like prometheus reporter, will get an incompatible metric name.

## Brief change log

  - override createLogicalScope in GenericMetricGroup

## Verifying this change

This change added tests and can be verified as follows:
  - run `MetricGroupTest#testLogicalScopeShouldIgnoreValueGroupName`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java [hunk 1]
```diff
@@ -216,6 +219,28 @@ public void testNameCollisionAfterKeyValueGroup() {
...
 
+	/**
+	 * Verifies that calling {@link AbstractMetricGroup#getLogicalScope(CharacterFilter, char, int)} on {@link GenericValueMetricGroup}
+	 * should ignore value as well.
+	 */
+	@Test
+	public void testLogicalScopeShouldIgnoreValueGroupName() {
+		Configuration config = new Configuration();
+		config.setString(ConfigConstants.METRICS_REPORTER_PREFIX + "test." + ConfigConstants.METRICS_REPORTER_CLASS_SUFFIX, TestReporter.class.getName());
+		MetricRegistryImpl registry = new MetricRegistryImpl(MetricRegistryConfiguration.fromConfiguration(config));
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java]: should  be wrapped in a try-catch block

File: flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java [hunk 2]
```diff
@@ -216,6 +219,28 @@ public void testNameCollisionAfterKeyValueGroup() {
...
+		GenericMetricGroup root = new GenericMetricGroup(registry, new DummyAbstractMetricGroup(registry), "root");
+
+		String key = "key";
+		String value = "value";
+
+		MetricGroup group = root.addGroup(key, value);
+
+		String logicalScope = ((AbstractMetricGroup) group)
+			.getLogicalScope(new DummyCharacterFilter(), registry.getDelimiter(), 0);
+		assertTrue("Key is missing from logical scope.", logicalScope.contains(key));
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java]: use hamcrest matchers instead:
```
assertThat(logicalScope, containsString(key));
assertThat(logicalScope, not(containsString(value)));
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java (hunk 1 above)
Comment (tony810430) [reply flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java]: Do you mean `registry` should be shutdown in `finally` clause?

Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java]: correct

**Comment by tony810430** [comment]
@zentol 

Have already addressed the comments.

**Pull request merged**